### PR TITLE
[JS] Topic : Template Strings in JavaScript

### LINKED ISSUE #999: JS: Template Strings in JavaScript
# Template Strings

Describe Template Strings in JavaScript.

# Directory
```
Topics/Template Strings
```
# Description
<li> Tell About Template Strings in JavaScript.
<li> Write about String Interpolation.
<li> Write about Tagged Templates.
<li> About Raw Strings.
<li> Examples & Syntax of Each.

# Formats

- [x] Document

### PR DESCRIPTION
**Pull request by AbhinandanAdhikari**
<hr>

## Description 📜

Created a tutorial for Template Strings in JavaScript.
Fixes #999 

<hr>

## Type of change 📝

- [ ] Audio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Video (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev (Flutter, Java, Kotlin)
- [ ] Backend Dev (Java, .NET, Python)
- [ ] C/CPP
- [ ] Competitive Programming
- [ ] Cyber Security
- [ ] DSA
- [ ] Database
- [ ] Datascience with Python
- [ ] Datascience with R
- [ ] Frontend Dev (HTML, CSS, JS)
- [ ] Frontend Dev (React, Angular, Vue)
- [ ] Golang
- [ ] Interview Prep
- [ ] Java
- [x] JavaScript
- [ ] MERN
- [ ] Machine Learning
- [ ] Open Source
- [ ] Python
- [ ] Rust
- [ ] Statistics
- [ ] UI/UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by AbhinandanAdhikari** [comment]
@paramsiddharth @dakshsethi @Creatoon @s4-2 Supervisor & Mentors! Please review my pull request.

**Comment by AbhinandanAdhikari** [comment]
@paramsiddharth Sir, please review this Pull Request and add a label to it.

**Review by Creatoon — CHANGES_REQUESTED** [review]

File: JavaScript/Topics/10.Template_Strings/README.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+Hello GWOC
+Hello GWOC
+```
+__Explanation:__
+
+Before ES6, we have to use ```+``` for string interpolation.
+
+# Tagged Templates
+
+Tagged Templates are allows to parse a Template literals with a function
```
[the remark this answers is not in the document]
Comment (AbhinandanAdhikari) [reply JavaScript/Topics/10.Template_Strings/README.md]: @paramsiddharth @Creatoon @dakshsethi @s4-2 Made the changes, please review it.

**Comment by paramsiddharth** [comment]
@dakshsethi @s4-2 Please review.

File: JavaScript/Topics/10.Template_Strings/README.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+__Example:__
+```javascript
+// with Template Literal
+
+console.log(`Welcome Everyone
+To GWOC`);
+
+// without Template Literal
+
+console.log("Welcome Everyone \n To GWOC");
```
Comment (paramsiddharth) [inline JavaScript/Topics/10.Template_Strings/README.md:38]: The padding doesn't match the literal.
```suggestion
console.log("Welcome Everyone\nTo GWOC");
```

File: JavaScript/Topics/10.Template_Strings/README.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    return str;
+}
+let res = tag`Hello, GWOC!`;
+console.log(res);
+```
+
+__Output:__
+```
+[ Hello, GWOC! ]
+```
```
Comment (paramsiddharth) [inline JavaScript/Topics/10.Template_Strings/README.md:110]: Please also include a well-known of tagged templates. :)

**Review by paramsiddharth — CHANGES_REQUESTED** [review]

**Comment by AbhinandanAdhikari** [comment]
@paramsiddharth @dakshsethi  @Creatoon @s4-2 Made the changes for this Pull Request, please review it.

**Review by dakshsethi — APPROVED** [review]
Looks good to me!! 🔥🔥
@paramsiddharth @Creatoon @s4-2 please review and approve.

**Comment by AbhinandanAdhikari** [comment]
@Creatoon @paramsiddharth Sir please review.

**Review by paramsiddharth — APPROVED** [review]

**Comment by paramsiddharth** [comment]
@Creatoon @s4-2 Please review.

**Pull request merged**